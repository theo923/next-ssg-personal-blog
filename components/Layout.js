import React from 'react';
import Navigation from './Navigation';
import Content from './Content';

const navItem = ['Blog', 'Project', 'About']
const catergories = ['React', 'NextJS', 'JavaScript', 'HTML5', 'CSS', 'Lifestyle', 'Music']

const Layout = ({item}) => {
    return (
        <div>
            <Navigation navItem={navItem} />
            <Content item={item} catergories={catergories} />
        </div>
    )
    
}

export default Layout;
