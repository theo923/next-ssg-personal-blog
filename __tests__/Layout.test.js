import { shallow } from "enzyme";
import React from "react";
import Layout from "../components/Layout";

describe("Layout test", () => {
  const setup = (props = {}, state = null) => {
    return shallow(<Layout {...props} />);
  };

  const findJSXByAttr = (name, wrapper) => {
    return wrapper.find(`[data-test="${name}"]`);
  };

  it("expect Layout component is rendered without crashing", () => {
    const wrapper = setup();
  });

  it("expect component-layout is rendered", () => {
    const wrapper = setup();
    const layout = findJSXByAttr("component-app-layout", wrapper);
    expect(layout.length).toBe(1);
  });
});
