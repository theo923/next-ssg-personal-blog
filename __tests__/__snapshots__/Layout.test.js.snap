// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout test expect Layout component is rendered && match with the Snapshot 1`] = `
<div>
  <Navigation
    navItem={
      Array [
        "Blog",
        "Project",
      ]
    }
  />
  <Content
    categories={
      Array [
        "React",
        "NextJS",
        "JavaScript",
        "HTML5",
        "CSS",
        "Lifestyle",
        "Music",
      ]
    }
  />
  <Footer />
</div>
`;
